frontend/packages/knative-plugin/src/components/test-function/modal-components/RequestOptions.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { Button, TextInputTypes } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { FormikProps, FormikValues } from 'formik/dist/types';
+import { useTranslation } from 'react-i18next';
+import { AsyncComponent } from '@console/internal/components/utils/async';
+import { DropdownField, InputField } from '@console/shared/src/components';
+import { FormLayout } from '@console/shared/src/components/cluster-configuration';
+import { InvokeFormat } from '../../../types';
+import '../TestFunctionModal.scss';
+
+const NameValueEditorComponent = (props) => (
+  <AsyncComponent
+    loader={() =>
+      import('@console/internal/components/utils/name-value-editor').then((c) => c.NameValueEditor)
+    }
+    {...props}
+  />
+);
+
+const RequestOptions: React.FC<FormikProps<FormikValues>> = ({ setFieldValue, values }) => {
+  const { t } = useTranslation();
+  const [showCustomHeaders, setShowCustomHeaders] = React.useState(false);
+  const {
+    request: {
+      headers: {
+        [InvokeFormat.HTTP]: { custom: customHTTPHeaders },
+        [InvokeFormat.CloudEvent]: { custom: customCloudEventHeaders },
+      },
+      format: invokeFormat,
+    },
+  } = values;
+
+  const handleNameValuePairs = ({ nameValuePairs: updatedNameValuePairs }) => {
+    /* Removing the extra element from arrays that are auto-generated */
+    updatedNameValuePairs.forEach((arr) => arr.length >= 3 && arr.splice(2, 1));
+
+    setFieldValue(`request.headers.${invokeFormat}.custom`, updatedNameValuePairs);
+  };
```

Comment:
Let us drop this magic code and add these "extra elements" when creating the request.

Reply:
Actually, due to this extra element, the `NameValueEditorComponent` is getting broken, since it needs only 2 elements.